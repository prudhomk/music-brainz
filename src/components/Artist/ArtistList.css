.ArtistList {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ArtistList li {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 250px;
  height: auto;
  cursor: pointer;
  list-style: none;
  margin: 5px;
  padding: 5px;
  border: 2px solid black;
  box-shadow: 2px 2px 10px 2px;
  transition: width 0.5s;
}

.ArtistList li:hover {
  background-color: pink;
  width: 300px;
}