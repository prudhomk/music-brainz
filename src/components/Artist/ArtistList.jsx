import React from 'react';
import Artist from './Artist';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


export default function ArtistList({ artistList }) {
  if(artistList) {
    const artistResult = artistList.map((artist) => (
      <>
        <li key={artist.id}>
          <Link to={`/${artist.id}/releases`}>
            <Artist {...artist}/>
          </Link>
        </li>;
      </>
    ));
  
    return (
      <>
        <ul>
          {artistResult} 
        </ul>
      </>
    );
  }
  return <h1>No results found</h1>;
}

ArtistList.propTypes = {
  artistList: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      country: PropTypes.string.isRequired,
    })
  )
};

