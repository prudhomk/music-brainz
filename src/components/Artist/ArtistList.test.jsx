import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import ArtistList from './ArtistList';

describe('Main display component snapshot', () => {
  it('render ArtistList', () => {
    const list = {
      name: 'name',
      country: 'country'
    };
    act(() => { 
      const { asFragment } = render(
        <ArtistList {...list} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
