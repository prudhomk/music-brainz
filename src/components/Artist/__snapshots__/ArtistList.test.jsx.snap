// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main display component snapshot render ArtistList 1`] = `
<DocumentFragment>
  <h1>
    No results found
  </h1>
</DocumentFragment>
`;
